### PR TITLE
fix gui internal daemon: stop gracefully

### PR DESCRIPTION
**Pull request by edouardparis**

### CODE REVIEW

File: gui/src/daemon/mod.rs [hunk 1]
```diff
@@ -43,11 +43,9 @@ impl std::fmt::Display for DaemonError {
 
 pub trait Daemon: Debug {
     fn is_external(&self) -> bool;
-    fn load_config(&mut self, _cfg: Config) -> Result<(), DaemonError> {
-        Ok(())
-    }
     fn config(&self) -> Option<&Config>;
-    fn stop(&mut self) -> Result<(), DaemonError>;
+    fn stop(&self);
+    fn is_stopped(&self) -> bool;
```
Comment (darosior) [inline gui/src/daemon/mod.rs:48]: This doesn't seem to be used anywhere?

File: gui/src/daemon/embedded.rs [hunk 2]
```diff
@@ -40,71 +30,36 @@ impl Daemon for EmbeddedDaemon {
...
 
-    fn stop(&mut self) -> Result<(), DaemonError> {
-        if let Some(h) = self.handle.take() {
-            let handle = h.into_inner().unwrap();
-            handle.shutdown();
+    fn stop(&self) {
+        self.handle.trigger_shutdown();
+        while !self.handle.shutdown_complete() {
+            tracing::debug!("Waiting daemon to shutdown");
+            std::thread::sleep(std::time::Duration::from_secs(1));
```
Comment (darosior) [inline gui/src/daemon/embedded.rs:41]: nit, let's avoid triggering the "process stuck"
```suggestion
            tracing::debug!("Waiting for daemon to shutdown");
            std::thread::sleep(std::time::Duration::from_millis(500));
```

File: gui/src/daemon/mod.rs (hunk 1 above)
Comment (edouardparis) [reply gui/src/daemon/mod.rs:48]: You are right, I am removing it

File: gui/src/daemon/embedded.rs [hunk 3]
```diff
@@ -10,22 +9,13 @@ use liana::{
 
 pub struct EmbeddedDaemon {
     config: Config,
-    handle: Option<RwLock<DaemonHandle>>,
+    handle: DaemonHandle,
```
Comment (darosior) [inline gui/src/daemon/embedded.rs:12]: Neaaat.

**Review by darosior — APPROVED** [review]
LGTM

**Comment by darosior** [comment]
Let's merge #625 before you make this point to master?

**Pull request merged**